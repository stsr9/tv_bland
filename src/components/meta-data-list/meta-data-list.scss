.data-row {
  @media (max-width: 800px) {
      flex-direction: row;
      &.icon-view {
        flex-direction: column;
      }
      &.wrap-row {
        display: unset;
        > * {
          width: 45%;
          float: left;
        }
      }
    }
  > * {
    border-bottom: 1px solid #ccc;
    padding: 15px 0;
    display: flex;
    @media (max-width: 800px) {
      flex-direction: column;
      border: unset;
      span {
        color: #ccc;
      }
    }
  }
  .with-icon {
    align-items: center;
    width: 100%;
    @media (max-width: 800px) {
      flex-direction: row;
    }
    i {
      font-size: 2em;
      margin-right: 20px;
      position: absolute;
      color: #ccc;
      @media (max-width: 800px) {
        font-size: 3em;
      }
    }
    div {
      padding-left: 60px;
      display: flex;
      flex-grow: 2;
      @media (max-width: 800px) {
        flex-direction: column;
      }
    }
  }
  label {
    font-weight: bold;
    flex-basis: 40%;
    margin-right: 30px;
  }
}
