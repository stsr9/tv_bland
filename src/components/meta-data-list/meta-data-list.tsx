import React from 'react'
import './meta-data-list.scss'

type Props = {
	icon?: string
	wrapRows?: boolean
	children: React.ReactNode[]
}

export const MetaDataList = ({ icon, children, wrapRows }: Props) => {
	return (
		children && (
			<div
				className={`data-row flex-column ${wrapRows ? 'wrap-row' : ''} ${
					icon ? 'icon-view' : ''
				}`}
			>
				{children.map((child: React.ReactNode, index: number) => {
					return icon ? (
						<div key={index} className="with-icon flex">
							<i className={`fa fa-${icon}`} />
							{child}
						</div>
					) : (
						child
					)
				})}
			</div>
		)
	)
}
