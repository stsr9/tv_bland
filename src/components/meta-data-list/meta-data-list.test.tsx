import renderer from 'react-test-renderer';
import { MetaDataList } from './meta-data-list';

it('renders without crashing', () => {
  const episodeDetails = renderer.create(
      <MetaDataList icon="user-circle">
          <div><label>Streamed on</label><span>Netflix</span></div>
           <div><label>Schedule</label><span>Monday</span></div>
      </MetaDataList>
  ).toJSON();

  expect(episodeDetails).toMatchSnapshot();
});
