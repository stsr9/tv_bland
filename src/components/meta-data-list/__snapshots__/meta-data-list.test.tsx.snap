// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="data-row flex-column  icon-view"
>
  <div
    className="with-icon flex"
  >
    <i
      className="fa fa-user-circle"
    />
    <div>
      <label>
        Streamed on
      </label>
      <span>
        Netflix
      </span>
    </div>
  </div>
  <div
    className="with-icon flex"
  >
    <i
      className="fa fa-user-circle"
    />
    <div>
      <label>
        Schedule
      </label>
      <span>
        Monday
      </span>
    </div>
  </div>
</div>
`;
